### PR TITLE
Add the deep fragmenter record-building interface

### PR DESCRIPTION
**Pull request by aileen**
Adds the deep fragmenter record-building interface from #220 without activating it in the CLI or Netlify adapters. Both adapters stay on the deprecated wrappers, so this change is dormant until the follow-up adoption slices.

## What changed

`@tryghost/algolia-fragmenter` gains one new public synchronous interface:

```ts
createAlgoliaRecords(ghostContent, options?): readonly AlgoliaRecord[]
```

- **Ghost content projection** — with no `contentProjection`, every record repeats `image`, `tags`, `authors`, and Ghost's computed `excerpt`. A supplied projection is the complete optional field set (empty allowed) drawn from the approved 14-source allowlist, with one flat validated alias per field and the approved collision rules (protected fields, the `customRanking` container, canonical allowlist names, Algolia-reserved names).
- **Ranking siblings** — `customRanking.heading` and `customRanking.position` stay package-owned; a projection may add siblings sourced only from `featured` and `reading_time`, each under a validated alias; missing values become `null`.
- **Whole-batch preflight** — policy, content, and size validation completes before any records are returned. Any deterministic problem throws one `FragmenterError` (`INVALID_POLICY`, `INVALID_GHOST_CONTENT`, or `RECORD_TOO_LARGE`) carrying all ordered issues; no partial array is ever returned.
- **Deterministic record size** — after legacy anchor grouping, whole extraction fragments are greedily packed so every complete compact record is at most 9,999 UTF-8 bytes (`Buffer.byteLength(JSON.stringify(record), 'utf8')`). The first record of a group keeps `<content id>_<group index>`; continuations append stable numeric suffixes; projection, heading, anchor, URL, and rank context are repeated in every record. An indivisible oversized fragment fails locally with actionable size context instead of truncating.
- **Fallback record** — Ghost content with no extraction fragments emits one deterministic projection-only record (empty `html`, no anchor, headingless rank) so the content stays discoverable.
- **Ignored slugs** — removed before full content/size validation, while every item must still carry a valid slug for the exclusion decision.

The deprecated `transformToAlgoliaObject` and `fragmentTransformer` wrappers keep their exact current behavior and do not gain projection or packing. Their existing tests pass unedited.

## Release notes (for the release handoff)

- **User-visible behavior**: none until an adapter adopts the new interface. The published package adds `createAlgoliaRecords`, `FragmenterError`, and the projection types as new exports; existing exports are unchanged.
- **Defaults**: the new interface defaults to projecting `image`, `tags`, `authors`, `excerpt`; `excerpt` is additive display text and never replaces fragment `html`. The wrappers keep their existing defaults.
- **Configuration/environment**: no CLI or Netlify configuration is read yet. The JSON-serializable `contentProjection` shape introduced here is the one later slices wire to CLI config and `ALGOLIA_CONTENT_PROJECTION`.
- **Affected records/settings**: none in production — no adapter calls the new interface. Projection never changes Algolia index settings.
- **Compatible versions**: consumes `@tryghost/algolia-html-extractor` `workspace:^` (0.1.0). CLI/Netlify releases selected by the scoped Nx dry run are dependency-only patches with no direct behavior change.
- **Migration order**: release as a fragmenter minor before #221 (CLI adoption) and #222 (Netlify adoption); consumers stay on the wrappers until those slices.
- **Source issues**: delivery slice of #220 under PRD #213; contributes to (but does not close) #43 and #148.
- **Rollback**: consumers remain on the wrappers, so rollback requires no data migration — roll forward with a patch or keep adapters on the previous fragmenter version.

## Verification

- `pnpm --filter @tryghost/algolia-fragmenter test` — 117 tests across 4 files pass, plus package typecheck, oxlint, and oxfmt (Node 24).
- `pnpm test` from the root — 263 tests across 21 files pass; coverage 97.05% statements / 95.00% branches / 99.36% functions / 96.98% lines against the 93/90/96/93 thresholds, with every fragmenter `src` module at 100% on all four measures.
- `pnpm typecheck` and `pnpm lint` — clean.
- The pre-existing `test/fragmenter.test.mts` and `test/compatibility.test.mts` pass unedited, proving the shared-grouping refactor changed no wrapper behavior. An independent review pass additionally diffed the previous built wrapper (`git show HEAD:.../lib/index.mjs`) against the rebuilt one across 134 edge-case comparisons (pre-tag merging, anchor grouping, duck-typed relations, ignore-slug shapes, accumulator seeding) with zero mismatches, including thrown-error identity.
- Byte-boundary tests are exact: one record accepted at 9,999 UTF-8 bytes and its neighbor rejected at 10,000 with `excess: 1`; multi-byte and JSON-escaping cases are constructed so a UTF-16 `length` measurement would produce the wrong record count.
- `test/package.acceptance.test.mts` now checks the packed tarball's 31-file `lib/` layout, the new runtime exports, a packed `createAlgoliaRecords` round-trip, a packed policy-failure throw, `FragmenterError` class identity, and declaration resolution from a clean temporary ESM consumer.
- `lib/` is regenerated build output with zero drift against a fresh `pnpm build`.

`CONTEXT.md` gains the glossary entries this slice introduces (ranking sibling, anchor group, fallback record, continuation record, record byte ceiling, preflight).

The release itself (scoped `pnpm ship` minor dry run, publish workflow, clean-install verification) stays with maintainers per the repository's release gates; note that #220 is nominally blocked by #219 (human-gated live-smoke census review), so merge/release ordering remains a maintainer call.

Closes nothing; tracks #220.

### CODE REVIEW

File: packages/algolia-fragmenter/README.md [hunk 1]
```diff
@@ -16,7 +16,59 @@ pnpm add @tryghost/algolia-fragmenter
...
+    ignoreSlugs: ['secret-page'],
+    contentProjection: {
+        fields: ['image', 'tags', {source: 'reading_time', as: 'readingMinutes'}],
+        customRanking: [{source: 'featured', as: 'isFeatured'}]
+    }
+});
+```
+
+- `ignoreSlugs` excludes content by slug before the rest of its fields are validated.
+- `contentProjection.fields` is the complete optional field set; it replaces the default set rather than patching it, and `[]` selects no optional fields. Without `contentProjection`, the optional fields are `image`, `tags`, `authors`, and `excerpt`.
```
[the remark this answers is not in the document]
Comment (aileen) [reply packages/algolia-fragmenter/README.md]: Fixed in a9ec979. The optional-fields bullet now states fields is required whenever contentProjection is supplied (verified: resolveContentProjection routes an absent fields into the invalid-shape branch, unlike customRanking which tolerates undefined), and the issue-context paragraph documents that index and the content id are null for the whole-batch shape issue and that the content id is null whenever the content has no usable id. Fragmenter suite passes.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_
